Expose wget_wch

lidavidm opened the pull request:
This is WIP, just wanted your thoughts on the implementation.

This exposes [`wget_wch`](http://linux.die.net/man/3/get_wch), which is like `getch` but gets a wide character. I've wrapped it in a more Rust-friendly API: the original function takes a pointer to the location to store the input and returns a result indicating whether the character is a keycode or a character. The exposed version returns an enum instead. If you like this approach, I can wrap the rest of the functions.

TODO:
- [x] Handle cbreak
- [x] Wrap rest of functions
- [x] Add example

Also, I noticed that the makefile doesn't quite work on my system, but Cargo can build the examples by simply adding them as binary targets. Would you be interested in a PR to convert to that way of building the examples?

jeaye commented:
Thanks for the WIP PR! I think what you have so far is dandy, but I'd like to ask why we should `panic!` instead of returning `None` or similar.

> Also, I noticed that the makefile doesn't quite work on my system, but Cargo can build the examples by simply adding them as binary targets. Would you be interested in a PR to convert to that way of building the examples?

Sure, ncurses-rs is definitely due for some cleanup.

lidavidm commented:
Yup, you're right. I've converted it to `Option<WchResult>` and also made the `KeyCode` case contain an `i32` to be consistent with the `*getch` functions. (So if there's an error, or if halfdelay mode is set, it'll return None, and you can pattern match on `KeyCode(KEY_F1)` without type errors.) I'll play with the cargo build in a separate PR (I'm just wondering if there's something the Makefile handles on certain platforms that cargo doesn't).

I've also added an example based on `ex_2.rs`.

jeaye commented:
Ok, great. Thanks so much for the work; is there anything else you want to do in this PR, or are we good to merge?

lidavidm commented:
I'm done here, I'll take a look at the Makefile and make a separate PR if I figure out something.

jeaye commented:
\o/
